test/callbacks/on_push_callback/1.callbacks.c

```diff
@@ -0,0 +1,39 @@
+#include <babeltrace2/babeltrace.h>
+#include <metababel/metababel.h>
+#include <stdio.h>
+
+struct s {
+  int counter;
+};
+
+void btx_initialize_usr_data(void *btx_handle, void **usr_data) {
+  *usr_data = calloc(1, sizeof(struct s));
+}
+
+void btx_finalize_usr_data(void *btx_handle, void *usr_data) { free(usr_data); }
+
+static void event_callback(void *btx_handle, void *usr_data,
```

Comment:
I think we need a test that considers two events, one of them has a callback to re-push messages, the other event just passes around. We expect `on_push` to be called the number of times `push_message` was called.

Reply:
Good idea, I can add that yes. It will also verify that it's called for begin / end